src/web/app/components/new-instructor-data-row/new-instructor-data-row.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { InstructorData } from './instructor-data';
+
+/**
+ * A single row of data of a new instructor.
+ */
+@Component({
+  // The following selector code style violation of https://angular.io/guide/styleguide#style-05-02 and
+  // https://angular.io/guide/styleguide#style-05-03 seems necessary according to
+  // https://stackoverflow.com/questions/55446740/how-to-add-row-component-in-table-in-angular-7
+  // tslint:disable-next-line:component-selector
+  selector: 'tr[tm-new-instructor-data-row]',
+  templateUrl: './new-instructor-data-row.component.html',
+  styleUrls: ['./new-instructor-data-row.component.scss'],
+})
+export class NewInstructorDataRowComponent implements OnInit {
+  @Input() instructor!: InstructorData;
+  @Input() index!: number;
+  @Input() activeRequests!: number;
+  @Output() addInstructorEvent: EventEmitter<void> = new EventEmitter();
+  @Output() removeInstructorEvent: EventEmitter<void> = new EventEmitter();
+  @Output() toggleEditModeEvent: EventEmitter<boolean> = new EventEmitter();
+
+  isBeingEdited: boolean = false;
+  editedInstructorName!: string;
+  editedInstructorEmail!: string;
+  editedInstructorInstitution!: string;
+
+  constructor() { }
+
+  ngOnInit(): void {
+    this.resetEditedInstructorDetails();
+  }
+
+  /**
+   * Adds the instructor at the i-th index.
```

Comment:
"i-th index" is not relevant in a row-specific component.
Also, this comment is redundant without it because "Adds the instructor" is parroting the method name.

Reply:
Okay, I think I will just remove this comment and the one for removing the instructor too.